File: src/main/scala/checker/BranchOp.java
package checker;

import java.util.Set;

public class BranchOp {
    public Set<Integer>[] domains;
    public BranchOp(Set<Integer>[] domains){ this.domains=domains;}

    @Override
    public BranchOp clone(){
        return new BranchOp(domains.clone());
    }

    @Override
    public String toString() {
        return "No more branching possible";
    }
}

File: src/main/scala/checker/incremental/BranchOp.scala
package checker.incremental

/**
  * Each instance of this class represents a branching
  * operation performed from given 'domains'.
  *
  * @param domains : array of domains.
  */
class BranchOp(var domains: Array[Set[Int]]) {
  override def clone = new BranchOp(domains.clone)

  override def toString = "No more branching possible"
}



